.github/workflows/integration_tests.yml

```diff
@@ -544,32 +565,31 @@ jobs:
           mv support-archive/keptn*.zip support-archive/${SUPPORT_ARCHIVE_FILENAME}.zip
 
       - name: Uninstall Keptn
+        if: always() && env.CLOUD_PROVIDER != 'GKE'
         timeout-minutes: 5
         run: echo "y" | keptn uninstall -n ${KEPTN_NAMESPACE}
 
-      - name: Uninstall remote execution plane
-        timeout-minutes: 5
-        if: env.REMOTE_EXECUTION_PLANE == 'true'
-        run: |
-          kubectl delete namespace keptn-helm-service
-          kubectl delete namespace keptn-jmeter-service
-
       - name: Uninstall Istio
-        if: env.CLOUD_PROVIDER != 'minishift-on-GHA' # istio was not installed on minishift, so no need to uninstall it
+        if: always() && env.CLOUD_PROVIDER != 'minishift-on-GHA' && env.CLOUD_PROVIDER != 'GKE' # istio was not installed on minishift or GKE, so no need to uninstall it
         timeout-minutes: 5
         run: test/utils/uninstall_istio.sh
 
-      - name: Cleanup GKE cluster
-        if: always() && env.CLOUD_PROVIDER == 'GKE' # we always need to cleanup GKE clusters
-        timeout-minutes: 10
-        env:
-          GKE_VERSION: ${{ matrix.PLATFORM_VERSION }}
-          GCLOUD_SERVICE_KEY: ${{ secrets.GCLOUD_SERVICE_KEY }}
-          GCLOUD_PROJECT_NAME: ${{ secrets.GCLOUD_PROJECT_NAME }}
-          CLOUDSDK_COMPUTE_ZONE: "us-east1-b"
-          CLOUDSDK_REGION: "us-east1"
-          CLUSTER_NAME_NIGHTLY: ${{ steps.install_gke.outputs.CLUSTER_NAME_NIGHTLY }}
-        run: test/utils/gke_delete_cluster.sh
+      - name: Cleanup test namespace
+        if: always() && env.CLOUD_PROVIDER == 'GKE'
+        run: |
+          readarray -t namespaces <<< "$(kubectl get namespaces | awk '{ print $1 }' | grep ${{ env.KEPTN_NAMESPACE_PREFIX }})"
+
+          if [[ "${{ github.event_name }}" == 'schedule' && "${{ steps.test_aggregated.outcome }}" == 'failure' ]]; then
+            for namespace in "${namespaces[@]}"; do
+              echo "Annotating namespace $namespace with Janitor TTL of 3 days..."
+              kubectl annotate namespace "$namespace" janitor/ttl=3d
+            done
+          else
+            for namespace in "${namespaces[@]}"; do
+              echo "Deleting namespace $namespace ..."
+              kubectl delete namespace "$namespace" --wait=false
+            done
+          fi
```

Comment:
This steps takes care of the cleanup on GKE clusters.
It will fetch all namespaces fro the test run and then decide whether they are deleted immediately or keptn for a few days.
If the test run was successful, the namespaces are deleted right away.
If the test run had errors, the namespaces are kept for 3 days and then cleaned up by the `kube-janitor` operator.